src/order/chain.lean

```diff
@@ -219,3 +230,47 @@ let ⟨h₁, H⟩ := chain_closure_max_chain.is_chain.super_chain_succ_chain h i
   H.ne (chain_closure_max_chain.succ_fixpoint_iff.mpr rfl).symm
 
 end chain
+
+/-! ### Flags -/
+
+/-- The type of flags, aka maximal chains, of an order. -/
+structure flag (α : Type*) [has_le α] :=
+(carrier : set α)
+(chain' : is_chain (≤) carrier)
+(max_chain' : ∀ ⦃s⦄, is_chain (≤) s → carrier ⊆ s → carrier = s)
```

Comment:
Why not just use `is_max_chain`? In fact, why not make this a `subtype`?

Reply:
Because `is_max_chain` is a conjonction, so you might as well split it in two fields. It also means that adding the intermediate structure `chain`, the type of chains, will be more or less trivial. This is very much the same situation as for `order.ideal`, which is not just `subtype is_ideal`, and #13070 which makes `order.ideal` extend `lower_set`. 

Regarding `subtype`, the tendency is to move set-like structures away from `subtype` in favor of structures. See for example #12035